src/handlers/net/v2/NetStreamHandler.h

```diff
@@ -0,0 +1,252 @@
+#include "StreamHandler.h"
+#include <string>
+
+namespace visor::handler::net::v2 {
+
+using namespace visor::input::pcap;
+using namespace visor::input::dnstap;
+using namespace visor::input::mock;
+
+static constexpr const char *NET_SCHEMA{"packets"};
+
+namespace group {
+enum NetMetrics : visor::MetricGroupIntType {
+    Counters,
+    Cardinality,
+    TopGeo,
+    TopIps
+};
+}
+
+struct NetworkPacket {
+    PacketDirection dir;
+    pcpp::ProtocolType l3;
+    pcpp::ProtocolType l4;
+    size_t payload_size;
+    bool syn_flag;
+    pcpp::IPv4Address ipv4_in;
+    pcpp::IPv4Address ipv4_out;
+    pcpp::IPv6Address ipv6_in;
+    pcpp::IPv6Address ipv6_out;
+
+    NetworkPacket(PacketDirection dir, pcpp::ProtocolType l3, pcpp::ProtocolType l4, size_t payload_size, bool syn_flag)
+        : dir(dir)
+        , l3(l3)
+        , l4(l4)
+        , payload_size(payload_size)
+        , syn_flag(syn_flag)
+    {
+    }
+};
+
+class NetworkMetricsBucket final : public visor::AbstractMetricsBucket
+{
+
+protected:
+    mutable std::shared_mutex _mutex;
+
+    Cardinality _srcIPCard;
+    Cardinality _dstIPCard;
+
+    TopN<visor::geo::City> _topGeoLoc;
+    TopN<std::string> _topASN;
+    TopN<uint32_t> _topIPv4;
+    TopN<std::string> _topIPv6;
+
+    // total numPackets is tracked in base class num_events
+    struct counters {
+        Counter UDP;
+        Counter TCP;
+        Counter OtherL4;
+        Counter IPv4;
+        Counter IPv6;
+        Counter TCP_SYN;
+        Counter total_in;
+        Counter total_out;
+        Counter total_unk;
+        Counter total;
+        Counter filtered;
+        counters()
+            : UDP(NET_SCHEMA, {"udp"}, "Count of UDP packets")
+            , TCP(NET_SCHEMA, {"tcp"}, "Count of TCP packets")
+            , OtherL4(NET_SCHEMA, {"other_l4"}, "Count of packets which are not UDP or TCP")
+            , IPv4(NET_SCHEMA, {"ipv4"}, "Count of IPv4 packets")
+            , IPv6(NET_SCHEMA, {"ipv6"}, "Count of IPv6 packets")
+            , TCP_SYN(NET_SCHEMA, {"protocol", "tcp", "syn"}, "Count of TCP SYN packets")
+            , total_in(NET_SCHEMA, {"in"}, "Count of total ingress packets")
+            , total_out(NET_SCHEMA, {"out"}, "Count of total egress packets")
+            , total_unk(NET_SCHEMA, {"unknown_dir"}, "Count of total unknown direction packets")
+            , total(NET_SCHEMA, {"total"}, "Count of total packets matching the configured filter(s)")
+            , filtered(NET_SCHEMA, {"filtered"}, "Count of total packets that did not match the configured filter(s) (if any)")
+        {
+        }
+    };
+    counters _counters;
+
+    Quantile<std::size_t> _payload_size;
+
+    Rate _rate_in;
+    Rate _rate_out;
+    Rate _rate_total;
+    Rate _throughput_in;
+    Rate _throughput_out;
+    Rate _throughput_total;
+
+    void _process_geo_metrics(const pcpp::IPv4Address &ipv4);
+    void _process_geo_metrics(const pcpp::IPv6Address &ipv6);
+
+public:
+    NetworkMetricsBucket()
```

Comment:
@rboucher-me I would like your help here to name the metrics

I've alredy fix the geo_loc, asn and removed "payload" metrics name

@weyrick the idea here is to add a new Net Handler with fix on metric names